include/LeMonADE/updater/UpdaterSimpleConnection.h

```diff
@@ -0,0 +1,176 @@
+along with LeMonADE.  If not, see <http://www.gnu.org/licenses/>.
+
+--------------------------------------------------------------------------------*/
+
+#ifndef LEMONADE_UPDATER_UPDATERSIMPLECONNECTION_H
+#define LEMONADE_UPDATER_UPDATERSIMPLECONNECTION_H
+
+
+#include <LeMonADE/updater/moves/MoveLocalBase.h>
+#include <LeMonADE/utility/RandomNumberGenerators.h>
+#include<LeMonADE/updater/AbstractUpdater.h>
+#include <LeMonADE/updater/moves/MoveConnectSc.h>
+
+/**
+ * @file
+ *
+ * @class UpdaterSimpleConnection
+ *
+ * @brief Simple simulation updater for generic polyrotaxanes.
+ *
+ * @details It takes the type of move as template argument MoveType
+ * and the number of mcs to be executed as argument for the constructor
+ *
+ * @tparam IngredientsType Ingredients class storing all system information( e.g. monomers, bonds, etc).
+ * @tparam MoveType name of the specialized move.
+ */
+
+template<class IngredientsType,class MoveType, class ConnectionMoveType>
+class UpdaterSimpleConnection:public AbstractUpdater
+{
+
+public:
+  /**
+   * @brief Standard Constructor initialized with ref to Ingredients and MCS per cycle
+   *
+   * @param ing a reference to the IngredientsType - mainly the system
+   * @param steps MCS per cycle to performed by execute()
+   */
+  UpdaterSimpleConnection(IngredientsType& ing,uint32_t steps = 1 )
+  :ingredients(ing),nsteps(steps){}
+  
+ 
+  /**
+   * @brief This checks all used Feature and applies all Feature if all conditions are met.
+   *
+   * @details This function runs over \a steps MCS and performs the moves.
+   * It setting the age of the system and prints a simple simple simulation speed
+   * in the number of attempted monomer moves per s (tried and performed monomer moves/s).
+   *
+   * @return True if function are done.
+   */
+  bool execute();
+
+  /**
+   * @brief This function is called \a once in the beginning of the TaskManager.
+   *
+   * @details It´s a virtual function for inheritance.
+   * Use this function for initializing tasks (e.g. init SDL)
+   *
+   **/
+  virtual void initialize();
+
+  /**
+   * @brief This function is called \a once in the end of the TaskManager.
+   *
+   * @details It´s a virtual function for inheritance.
+   * Use this function for cleaning tasks (e.g. destroying arrays, file outut)
+   *
+   **/
+  virtual void cleanup(){};
+
+private:
+  //! Specialized move to be used for the movement of the monomers 
+  MoveType move;
+  //! 
+  ConnectionMoveType connectionMove;
+  //! Number of mcs to be executed
+  uint32_t nsteps;
+  //! random number generator (seed set in main program)
+  RandomNumberGenerators rng;
+  //! 
+  IngredientsType& ingredients;
+  //! number of monomers which could have a reaction bond
+  uint32_t NReactiveSites;
+  //! number of already reacted monomers
+  uint32_t NReactedSites;
+  //! get conversion of the reaction process
+  double getConversion(){return (1.*NReactedSites)/(1.*NReactiveSites);};
```

Comment:
Conversion is defined as c= num all established Links / num all maximum Links
Does this calculated a value between [0;1]?

Reply:
This return an "undefined" values as the implementation is not correct due to the "counting" of NReactiveSites. Just try this with an arbitrary system.

Reply:
found the bugs and correct them see a0b7e7c0070cf3f82d6fe0752300617be96b2afb